### PR TITLE
edit the python files

### PR DESCRIPTION
**Pull request by willingc**
A quick pass through the Python code. Since you are supporting 3.6+ you may as well use f-strings.

### CODE REVIEW

File: pick_kernel/kernel.py [hunk 1]
```diff
@@ -240,7 +238,7 @@ def _publish_display_data(
             ident=self._topic("display_data"),
         )
 
-    async def launch_or_reuse_kernel(self, strema, ident, parent):
+    async def launch_or_reuse_kernel(self, stream, ident, parent):
```
Comment (rgbkrk) [inline pick_kernel/kernel.py:242]: 😅

File: pick_kernel/kernel.py [hunk 2]
```diff
@@ -380,6 +378,7 @@ def relay_to_kernel(self, stream, ident, parent):
 
 
 class PickyKernelApp(IPKernelApp):
+    """A kernel application for starting a `PickyKernel`, an IPython based kernel."""
```
Comment (rgbkrk) [inline pick_kernel/kernel.py]:
```suggestion
    """A kernel application for starting a `PickyKernel`, a proxying kernel with options."""
```

File: pick_kernel/kernelspec.py [hunk 3]
```diff
@@ -8,19 +8,18 @@
...
 
 from jupyter_client.kernelspec import KernelSpecManager
+from ipykernel import kernelspec as ipyks
 
 pjoin = os.path.join
 
-KERNEL_NAME = "picky-python%i" % sys.version_info[0]
-
-from ipykernel import kernelspec as ipyks
+KERNEL_NAME = f"picky-python{sys.version_info[0]}"
```
Comment (rgbkrk) [inline pick_kernel/kernelspec.py:15]: ❤️

File: pick_kernel/kernelspec.py [hunk 4]
```diff
@@ -103,12 +102,13 @@ def install(
         Whether to do a user-only install, or system-wide.
     kernel_name: str, optional
         Specify a name for the kernelspec.
-        This is needed for having multiple Picky kernels for different environments.
+        Names are needed to have multiple Picky kernels for different environments.
```
Comment (rgbkrk) [inline pick_kernel/kernelspec.py:105]: 👍

File: pick_kernel/kernel.py [hunk 5]
```diff
@@ -102,17 +96,19 @@ def __init__(self, *args, **kwargs):
...
         # TODO: Check errors from this loop and restart as needed (or shut down the kernel)
         loop.add_callback(self.relay_iopub_messages)
 
     async def relay_iopub_messages(self):
+        """Relay messages received by the Picky Kernel
+           to the consumer client (e.g. notebook)
+        """
         while True:
-            # Get message from our
             msg = await self.iosub.recv_multipart()
```
Comment (rgbkrk) [inline pick_kernel/kernel.py:111]:
```suggestion
            msg = await self.iosub.recv_multipart()
            # Send the message up to the consumer (for example, the notebook)
```

File: pick_kernel/kernel.py [hunk 6]
```diff
@@ -59,13 +52,14 @@ async def relay_shell(self):
 
 
 class PickyKernel(Kernel):
+    """A kernel adding the kernel magic and configuration of environments"""
```
Comment (rgbkrk) [inline pick_kernel/kernel.py]:
```suggestion
    """A kernel that accepts kernel magics which configure the environment"""
```

File: pick_kernel/kernel.py [hunk 7]
```diff
@@ -1,31 +1,24 @@
...
 # Rely on Socket subclass that returns Futures for recv*
 from zmq.eventloop.future import Context
 
-from jupyter_client import AsyncKernelManager
-from jupyter_client.session import extract_header
+from ipykernel.jsonutil import json_clean
 from ipykernel.kernelbase import Kernel
 from ipykernel.kernelapp import IPKernelApp
+from jupyter_client import AsyncKernelManager
+from jupyter_client.session import extract_header
```
Comment (rgbkrk) [inline pick_kernel/kernel.py:18]: Thank you for all the cleanup!

Comment (willingc) [reply pick_kernel/kernel.py:18]: No problem. Did it while watching a movie with Jake.

**Pull request merged**